plugins/blockvolume/blockprovider/gluster-virtblock/gluster_virtblock.go

```diff
@@ -99,21 +99,33 @@ func (g *GlusterVirtBlk) CreateBlockVolume(name string, size uint64, hostVolume
 		return nil, err
 	}
 
-	err = utils.ExecuteCommandRun("mkfs.xfs", "-f", blockFileName) //nolint: gosec
-	if err != nil {
-		logger.WithError(err).Errorf("failed to format block file " + blockFileName)
-		return nil, err
+	if blockVolOpts.BlkType != "none" {
+		fsType := blockVolOpts.BlkType
+		err = utils.ExecuteCommandRun(fmt.Sprintf("mkfs.%s", fsType), "-f", blockFileName) //nolint: gosec
+		if err != nil {
+			logger.WithError(err).Errorf("failed to format block file " + blockFileName + " with filesystem " + fsType)
+			return nil, err
+		}
 	}
 
 	resizeFunc := func(blockHostingAvailableSize, blockSize uint64) uint64 { return blockHostingAvailableSize - blockSize }
-	if err = blkUtils.ResizeBlockHostingVolume(hostVolume, size, resizeFunc); err != nil {
+	if err = hostvol.ResizeBlockHostingVolume(hostVolume, size, resizeFunc); err != nil {
 		logger.WithError(err).Error("failed in updating hostvolume _block-hosting-available-size metadata")
 		return nil, err
 	}
 
-	if err = clusterLocks.Lock(hostVolume); err != nil {
-		logger.WithError(err).Error("error in acquiring cluster lock")
-		return nil, err
+	for {
```

Comment:
infinite loop could cause an issue, which may lead to http timeout

Reply:
So we try for few iterations, and then return with LockTimeout? Then the csi should handle this error and retry?

Reply:
if volume creation fails, CSI will do a retry